### PR TITLE
Make Symfony 5.4 the minimum requirement

### PR DESCRIPTION
**Pull request by leofeyer**
This PR makes Symfony 5.4 the minimum requirement and fixes all deprecations except the security related ones, which @bytehead is going to fix in #3746.

### CODE REVIEW

File: core-bundle/README.md [hunk 1]
```diff
@@ -121,10 +121,10 @@ security:
                     - contao.security.logout_handler
 
     access_control:
-        - { path: ^%contao.backend.route_prefix%/login$, roles: IS_AUTHENTICATED_ANONYMOUSLY }
-        - { path: ^%contao.backend.route_prefix%/logout$, roles: IS_AUTHENTICATED_ANONYMOUSLY }
+        - { path: ^%contao.backend.route_prefix%/login$, roles: PUBLIC_ACCESS }
```
Comment (leofeyer) [inline core-bundle/README.md:124]: @bytehead @aschempp Is this correct?

File: core-bundle/src/Controller/Page/RootPageController.php [hunk 2]
```diff
@@ -39,8 +39,8 @@ private function getNextPage(int $rootPageId): PageModel
             return $nextPage;
         }
 
-        if ($this->has('logger')) {
-            $this->get('logger')->error(
+        if ($this->container->has('logger')) {
+            $this->container->get('logger')->error(
```
Comment (leofeyer) [inline core-bundle/src/Controller/Page/RootPageController.php:43]: I really think this should have been a constructor injection.

Comment (Toflar) [reply core-bundle/src/Controller/Page/RootPageController.php:43]: Everything should be constructor injection if it does not need to be conditionally loaded. E.g. every controller that uses "service A" all the time, should use constructor injection and never subscribe to it. You should only subscribe if you provide e.g. abstract functions other children *can* use. Just like `AbstractController` itself does. There are other ideas like `AbstractJobType` etc. But if the code needs the service, it should always use constructor injection.
Probably not in this PR though ;)

**Review by aschempp — APPROVED** [review]

File: core-bundle/README.md (hunk 1 above)
Comment (aschempp) [reply core-bundle/README.md:124]: Looks correct to me

**Review by Toflar — APPROVED** [review]

File: core-bundle/README.md (hunk 1 above)
Comment (bytehead) [reply core-bundle/README.md:124]: Yep, seems to be correct.

**Review by bytehead — APPROVED** [review]

**Pull request merged**